### PR TITLE
Feature/create character

### PR DESCRIPTION
**Pull request by antgreen13**
Add functionality to save a character

This change will use Flask to create a webservice call that will allow users to save a new character. The new character will be saved to that user's collection in Firebase.

### CODE REVIEW

File: src/app/home/home.component.ts [hunk 1]
```diff
@@ -53,6 +56,7 @@ export class HomeComponent implements OnInit, OnDestroy {
   public signOut(): void {
     this.authService.signOut();
     this.dataArray = [];
+    this.appCache.authorizedUser = undefined;
```
Comment (ericchen94) [inline src/app/home/home.component.ts:59]: Should we also set user property to undefined upon signOut?

File: src/app/creation/creation.component.html [hunk 2]
```diff
@@ -1,115 +1,156 @@
...
+          cdkDropList
+          [cdkDropListData]="str"
+          class="example-list"
+          (cdkDropListDropped)="drop($event)"
+        >
+          <div class="example-box" *ngFor="let item of str" cdkDrag>
+            {{ item }}
+          </div>
+        </div>
+      </div>
```
Comment (ericchen94) [inline src/app/creation/creation.component.html:64]: This is duplicated many times, can we use a ngFor to iterate through every ability instead?

File: firebase.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        character_data = db.collection(user_email)
+        if character_name:
+          character = character_data.document(character_name).get()
+          return flask.jsonify(character.to_dict()), 200
+        else:
+          all_characters = [doc.to_dict() for doc in character_data.stream()]
+          return flask.jsonify(all_characters), 200
+
+    except Exception as e:
+        return f"An Error Occurred: {e}"
```
Comment (ericchen94) [inline firebase.py]: Can make exception/error logging more descriptive

File: firebase.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+def save():
+  """
+    save() : Add document to Firestore collection with request body.
+    Ensure you pass a custom userEmail and characterName as part of json body in post request
+  """
+  if flask.request.method == 'POST':
+    user_email = flask.request.json['userEmail']
+    character_name = flask.request.json['name']
+    doc_ref = db.collection(user_email).document(character_name)
+    doc_ref .set(flask.request.json)
```
Comment (ericchen94) [inline firebase.py]: extra space here?

**Review by ericchen94 — CHANGES_REQUESTED** [review]

File: firebase.py (hunk 3 above)
Comment (antgreen13) [reply firebase.py]: Yes I'll make that change.

File: firebase.py (hunk 4 above)
Comment (antgreen13) [reply firebase.py]: Yeah there is. Weird that it still added that data correctly in firebase. I'll fix it,

File: src/app/home/home.component.ts (hunk 1 above)
Comment (antgreen13) [reply src/app/home/home.component.ts:59]: We don't have to, the auth service takes care of that variable.

File: src/app/creation/creation.component.html (hunk 2 above)
Comment (antgreen13) [reply src/app/creation/creation.component.html:64]: I think I would be able to do that. Right now I have it set up as a ticket in my project, and was planning on making that change as a part of my cleanup work at the end.

**Review by ericchen94 — APPROVED** [review]

**Pull request merged**